### PR TITLE
DMP-1657: Add /admin/user functional tests

### PR DESCRIPTION
**Pull request by chrisbellingham-hmcts**
## [DMP-1657](https://tools.hmcts.net/jira/browse/DMP-1657)

Add functional tests for `GET /admin/users` and `POST /admin/users/search`, the main additions are `shouldGetUsers()` and `shouldGetUsersBySearchCriteria()`.

Remaining changes are refactorings to reduce duplication and improve consistency of tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by hemantasharma1129 — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java [hunk 1]
```diff
@@ -98,22 +106,20 @@ void shouldModifyUser() {
...
+        int userId = extractId(createUser());
 
         Response getUserResponse = buildRequestWithExternalGlobalAccessAuth()
-            .baseUri(getUri(ADMIN_USERS))
             .contentType(ContentType.JSON)
-            .queryParam(COURTHOUSE_ID, 21)
-            .header(EMAIL_ADDRESS, "james.smith@hmcts.net")
-            .get()
+            .header(EMAIL_ADDRESS_HEADER_NAME, "james.smith@hmcts.net")
+            .queryParam("courthouse_id", 21)
```
Comment (mestebanez) [inline src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java:114]: For my own understanding where does courthouse 21 come from? Its not in the staging db? I guess this is not meant to exist?

Comment (mestebanez) [reply src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java:114]: Actually looking at this courthouse can be removed. Its not part of the endpoint spec.

File: src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java [hunk 2]
```diff
@@ -203,25 +236,36 @@ private Response createUser() {
...
-    @Test
-    void shouldGetSecurityGroups() throws JsonProcessingException {
+    private Response createUserWithSecurityGroups() {
         Response response = buildRequestWithExternalGlobalAccessAuth()
-            .baseUri(getUri("/admin/security-groups"))
             .contentType(ContentType.JSON)
-            .get()
+            .body("""
+                      {
+                           "full_name": "James Smith",
```
Comment (mestebanez) [inline src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java:252]: i am currently looking to make a proposal that we prefix all strings with func-. We can then have a general rule for devs not to create records with a prefix of func-. This should act as a belt and braces approach to stabilising our tests.

File: src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java (hunk 1 above)
Comment (chrisbellingham-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java:114]: Good spot, have removed.

**Pull request merged**